tests/behat/features/bootstrap/LibContext.php

```diff
@@ -0,0 +1,1691 @@
+  }
+
+  /**
+   * Implements step to check for a title to be present on the first page of a search result.
+   *
+   * @param string $title
+   *   The title to search for.
+   *
+   * @Then I can see :title in the search results first page
+   *
+   * @throws Exception
+   *   In case of errors.
+   */
+  public function findTitleInTheSearchResultsFirstPage($title) {
+    $title = $this->translateArgument($title);
+
+    $this->check($this->searchPage->findTitleOnPage($title));
+  }
+
+  /**
+   * GetPage - quick reference to the getPage element. Makes code more readable.
+   *
+   * @return \Behat\Mink\Element\DocumentElement
+   *   returns a page element
+   */
+  public function getPage() {
+    return $this->minkContext->getSession()->getPage();
+  }
+
+  /**
+   * Implements step to check for a login prompt.
+   *
+   * @Then I am prompted to login
+   *
+   * @throws Exception
+   *   In case of error.
+   */
+  public function getPromptToLogin() {
+     $this->check($this->objectPage->getPromptToLogin());
+  }
+
+  /**
+   * Navigate to a page.
+   *
+   * @todo should only navigate if the path is different from the current.
+   *
+   * @param string $path
+   *   The path to navigate to.
+   */
+  public function gotoPage($path) {
+    $this->minkContext->visit($path);
+  }
+
+  /**
+   * Go to the search page.
+   *
+   * @param string $string
+   *   String to search for.
+   *
+   * @Given I have searched for :string
+   *
+   * @throws Exception
+   *   In case of error.
+   */
+  public function gotoSearchPage($string) {
+    // First we try to translate the argument, to see if there's anything we should pick out first.
+    $searchString = $this->translateArgument($string);
+    $this->logMsg(($this->searchPage->getVerboseSearchResult() == "on"), "Searches for " . urlencode($searchString) . "\n");
+    $this->lastSearchString = $searchString;
+    $this->gotoPage('/search/ting/' . urlencode($searchString));
+  }
+
+
+  /**
+   * Implements step to search for data and put it into a file as known data.
+   *
+   * @Given I create files :mfile from opensearch on relation :relation
+   *
+   * @Param string $mfile
+   *    Filename basis for the two files being created, appended with _pos.mat and _neg.mat
+   * @Param string $relation
+   *    The relation we are looking for, in opensearch terminology, f.ex. dbcaddi:hasCreatorDescription
+   *
+   * @throws ErrorException
+   *   In case of error.
+   */
+  public function ICreateFilesForRelation($mfile, $relation) {
+    for ($i = 1; $i < 12000; $i = $i + 50) {
+      $this->ICreateFilesForRelationChunk($mfile, $relation, $i);
+    }
+  }
+
+  /**
+   * Helper function for ICreateFilesForRelation.
+   *
+   * @param string $mfile
+   *    Filename base for the two files being created.
+   * @param string $relation
+   *    The relation we are looking for.
+   * @param int $start
+   *    The starting point for the service retrieval.
+   *
+   * @throws ErrorException
```

Comment:
Redundant @throws

Reply:
Not redundant. The search using CURL may result in this exception. Changed the description to clarify this, but this is not redundant.